### PR TITLE
AIS-149: Add ability to override datatypes mapping in external table for PXF

### PR DESCRIPTION
**Pull request by RomaZe**

### CODE REVIEW

File: server/pxf-jdbc/src/main/java/org/greenplum/pxf/plugins/jdbc/JdbcPredicateBuilder.java [hunk 1]
```diff
@@ -39,6 +39,7 @@
 public class JdbcPredicateBuilder extends ColumnPredicateBuilder {
 
     private final DbProduct dbProduct;
+    private boolean convertOracleDate = false;
```
Comment (iamlapa) [inline server/pxf-jdbc/src/main/java/org/greenplum/pxf/plugins/jdbc/JdbcPredicateBuilder.java]: rename from `convertOracleDate` to `wrapDateWithTime `

File: server/pxf-jdbc/src/main/java/org/greenplum/pxf/plugins/jdbc/utils/DbProduct.java [hunk 2]
```diff
@@ -51,6 +61,15 @@ public String wrapDate(Object val) {
             return "to_date('" + val + "', 'YYYY-MM-DD')";
         }
 
+        @Override
+        public String wrapDateWithTime(Object val) {
+            int index = String.valueOf(val).lastIndexOf('.');
```
Comment (iamlapa) [inline server/pxf-jdbc/src/main/java/org/greenplum/pxf/plugins/jdbc/utils/DbProduct.java]: Refactoring:
```java
 String valStr =String.valueOf(val); 
 int index = valStr.lastIndexOf('.');
    if (index != -1) {
        valStr = valStr.substring(0, index);
    }
```

File: server/pxf-jdbc/src/main/java/org/greenplum/pxf/plugins/jdbc/JdbcPredicateBuilder.java (hunk 1 above)
Comment (RomaZe) [reply server/pxf-jdbc/src/main/java/org/greenplum/pxf/plugins/jdbc/JdbcPredicateBuilder.java]: Renamed

File: server/pxf-jdbc/src/main/java/org/greenplum/pxf/plugins/jdbc/utils/DbProduct.java (hunk 2 above)
Comment (RomaZe) [reply server/pxf-jdbc/src/main/java/org/greenplum/pxf/plugins/jdbc/utils/DbProduct.java]: Refactored

File: server/pxf-jdbc/src/main/java/org/greenplum/pxf/plugins/jdbc/JdbcBasePlugin.java [hunk 3]
```diff
@@ -134,6 +134,9 @@ public static TransactionIsolation typeOf(String str) {
     // Query timeout.
     protected Integer queryTimeout;
 
+    // Convert Postgres timestamp to Oracle date with time
+    protected Boolean wrapDateWithTime = false;
```
Comment (iamlapa) [inline server/pxf-jdbc/src/main/java/org/greenplum/pxf/plugins/jdbc/JdbcBasePlugin.java]: Use primitive

**Comment by iamlapa** [comment]
@RomaZe  pls update Readme.md

File: server/pxf-jdbc/src/main/java/org/greenplum/pxf/plugins/jdbc/JdbcBasePlugin.java (hunk 3 above)
Comment (RomaZe) [reply server/pxf-jdbc/src/main/java/org/greenplum/pxf/plugins/jdbc/JdbcBasePlugin.java]: replaced with primitive

**Comment by RomaZe** [comment]
Updated README.md

**Review by iamlapa — APPROVED** [review]

**Pull request merged**